File: ngPlanter/src/src/app/planter.ts
import { Plant } from "./plant";

export class Planter {
    userId: string;
    plantId: string;
    plantName: string;
    sensorDataList: SensorData[];
    
  
    constructor(userId: string, plantId: string, plantName: string, sensorDataList: SensorData[]) {
      this.userId = userId;
      this.plantId = plantId;
      this.plantName = plantName;
      this.sensorDataList = sensorDataList;
    }
}

export class SensorData {
    sensorName: string;
    sensorValue: number;
  
    constructor(sensorName: string, sensorValue: number) {
      this.sensorName = sensorName;
      this.sensorValue = sensorValue;
    }
}

File: IOT_Pet_Planter/src/main/main/java/com/Whitecape/PFE/models/Planter.java
package com.Whitecape.PFE.models;

import java.util.List;
import java.util.Base64;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "Planter")
public class Planter {
    @Id
    private String planterId;
    private String userId;
    private String plantId;
    private String plantName;
    private byte[] imageUrl;

	private List<SensorData> sensorDataList;
	public Planter() {
	}
    public byte[] getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(byte[] bytes) {
		this.imageUrl = bytes;
	}


    public Planter(String userId,String plantName, String plantId, List<SensorData> sensorDataList,byte[] imageUrl) {
        this.userId = userId;
        this.plantName = plantName;
        this.plantId = plantId;
        this.imageUrl=imageUrl;
        this.sensorDataList = sensorDataList;
    }

	public String getPlanterId() {
		return planterId;
	}

	public void setPlanterId(String planterId) {
		this.planterId = planterId;
	}

	public String getUserId() {
		return userId;
	}

	public void setUserId(String userId) {
		this.userId = userId;
	}

	public String getPlantId() {
		return plantId;
	}

	public void setPlantId(String plantId) {
		this.plantId = plantId;
	}

	public List<SensorData> getSensorDataList() {
		return sensorDataList;
	}

	public void setSensorDataList(List<SensorData> sensorDataList) {
		this.sensorDataList = sensorDataList;
	}


	public void setPlantName(String plantName) {
		this.plantName = plantName;
	}
	public String getPlantName() {
		return plantName;
	}
	public String getImageUrlAsBase64() {
		if (imageUrl != null && imageUrl.length > 0) {
			return Base64.getEncoder().encodeToString(imageUrl);
		}
		return null;
	}
}

